### PR TITLE
Features/show marker

### PR DESCRIPTION
**Pull request by GrandSuccess87**
User is currently able to fill a square with a 'X' marker.  I passed props from the parent board component to the square component (child) to do this.  I also cleaned up my code and removed unnecessary code.

### CODE REVIEW

File: src/components/gameComponent/gameComponent.js [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
-//                 {/* //based on state variable the board will show */}
-//                     {this.state.show && <Board /> } 
-//                 </div>
-//             </div>
-//         );
-//     }
+        constructor(props) {
+        super(props) // call super parent class constructor props
+        this.state = {
+            show: false // if true the data will be shown
```
Comment (soft-circles) [inline src/components/gameComponent/gameComponent.js]: Comments aren't necessary here. Commenting is helpful when it's not clear WHY something is doing something. In this case, `show: false` seems pretty self-explanatory.

File: src/components/gameComponent/gameComponent.js [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
+    }
 
-// }
+    toggleDiv = (e) => {
+        e.preventDefault();
+        console.log("Start Button Clicked!");
+        const { show } = this.state;
+        this.setState({show: !show}) 
+        // this.setState = { show: !show} // change values in state attribute
+        console.log(this.state);
```
Comment (soft-circles) [inline src/components/gameComponent/gameComponent.js]: as a general rule, make sure to skim your branch for any console logs and remove them before pushing up. We should never be pushing debugging console logs to `master`

File: src/components/squareComponent/squareComponent.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            showMarker: true, // if true then marker will be shown  
+            xValue: null  
+        } 
+    }
+
+    toggleSquare = (e) => {
+        e.preventDefault();
+        const { showMarker } = this.state;
+        this.setState({showMarker});
+        console.log("Square Button Clicked!");
```
Comment (soft-circles) [inline src/components/squareComponent/squareComponent.js]: remove console log

File: src/components/squareComponent/squareComponent.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        } 
+    }
+
+    toggleSquare = (e) => {
+        e.preventDefault();
+        const { showMarker } = this.state;
+        this.setState({showMarker});
+        console.log("Square Button Clicked!"); 
+        this.setState({xValue: 'X'})
+        console.log(this.state);
```
Comment (soft-circles) [inline src/components/squareComponent/squareComponent.js]: remove console log

File: src/components/squareComponent/squareComponent.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    constructor(props) {
+        super(props) // call super parent class constructor props
+        this.state = {
+            showMarker: true, // if true then marker will be shown  
+            xValue: null  
+        } 
+    }
+
+    toggleSquare = (e) => {
```
Comment (soft-circles) [inline src/components/squareComponent/squareComponent.js:12]: Where is the test(s) for this method?

**Review by soft-circles — CHANGES_REQUESTED** [review]
Nice first PR!
Please remember to remove any `console.logs` from your branch when you push up. We never want to merge debugging logs into `master`.

Also, remember we want to develop in the TDD style, so please write tests before writing your production code. Write a test, watch it fail, make it go green, and then refactor as needed.

File: src/components/squareComponent/squareComponent.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                <div>
+                    <button type="button" onClick={this.toggleSquare} className="btn btn-outline-primary board-Square">{this.state.xValue}</button> 
+
+                </div>
+
+            </div>
+
+        );
+    }
+
```
Comment (tamminhdao) [inline src/components/squareComponent/squareComponent.js:28]: remove extra line like this one

**Review by tamminhdao — CHANGES_REQUESTED** [review]
You should look into squashing all your commits into one to keep a clean commit history. It's a pretty small feature and commit like "remove unnecessary code" doesn't need to be in the log.

File: src/components/squareComponent/squareComponent.js (hunk 5 above)
Comment (GrandSuccess87) [reply src/components/squareComponent/squareComponent.js:12]: Hey Andy, thanks for the great feedback, I'm confused on how to test state in React..should this be in an expect() toEqual()..?

File: src/components/squareComponent/squareComponent.js (hunk 6 above)
Comment (GrandSuccess87) [reply src/components/squareComponent/squareComponent.js:28]: Hey Tam, thanks for the feedback! When you say squashing all commits into one, do you mean perform a rebase of my commits?

File: src/components/squareComponent/squareComponent.js (hunk 5 above)
Comment (soft-circles) [reply src/components/squareComponent/squareComponent.js:12]: I haven't tested state in react in a while, but maybe this blog post might help? https://medium.com/@newyork.anthonyng/testing-react-components-state-b57bfc712b90

Comment (GrandSuccess87) [reply src/components/squareComponent/squareComponent.js:12]: I currently have the test to pass which will now render the board when the start button is clicked and will default to no board shown when it is not clicked. Should now be able to merge with master..hopefully

**Comment by GrandSuccess87** [comment]
Removed debugging logs from code. Followed TDD cycle to get tests to pass.

**Comment by GrandSuccess87** [comment]
Hi Tam, 

Is it okay to merge this with master?
Would you like me to look into rebasing my commits?

**Comment by tamminhdao** [comment]
Yes, follow the link Andy sent on Slack and see if you can squash these into more compact commits.

**Pull request closed without merging**